### ISSUE TITLE
Use my own compute_loss with RewardTrainer

### ISSUE THREAD

**Issue by allzero-kwon** [lvwerra/trl#535]
Hi. I want to use my own RewardModel (not inherited from AutoModelForSequenceClassification)

But, i got message like below.
```
compute_loss is only implemented for RewardDataCollatorWithPadding, please implement your own compute_loss method if you are using a custom data collator
```

Is there any parameter which i can pass my own compute_loss to RewardTrainer ?

**Comment by lvwerra** [comment]
Hi @allzero-kwon, it looks like you passed a custom data collator which is why you get this message. @younesbelkada maybe we don't need to through an error here, no? since a user might pass a viable alternative to the default collator that does not require writing a custom loss?

In any case, if you want to write your own loss function you can do it as follows:
```python
from trl import RewardTrainer
class MyRewardTrainer(RewardTrainer):
    def compute_loss(self, model, inputs):
        labels = inputs.pop("labels")
        outputs = model(**inputs)
        logits = outputs[0]
        return my_custom_loss(logits, labels)
``` 

Hope this helps!

**Comment by allzero-kwon** [comment]
Thanks !
